### ISSUE TITLE
When there is a OOM the heap is dumped as `$(pwd)`

### ISSUE THREAD

**Issue by TharmiganK** [ballerina-platform/ballerina-lang#37079]
**Description:**

When there is an Out Of Memory issue with ballerina, the heap dumps are created as `$(pwd)` file without extension and renaming them as `<name>.hprof` is giving the actual heap dump.

**Steps to reproduce:**

Observed during the following OOM issues : 
1. https://github.com/ballerina-platform/ballerina-lang/issues/37075

2. https://github.com/wso2-enterprise/internal-support-ballerina/issues/132

**Sample log :**
```
Compiling source
        ballerina/http_tests:2.4.0
java.lang.OutOfMemoryError: Java heap space
Dumping heap to $(pwd) ...
Heap dump file created [1766375770 bytes in 17.196 secs]
```

**Comment by nirmal070125** [comment]
Can this be prioritized and fix?

**Comment by TharmiganK** [comment]
@nirmal070125 This issue has been addressed via https://github.com/ballerina-platform/ballerina-lang/issues/40547 and the fix will be available in `2201.7.0`. 
Adding @NipunaMadhushan for more information
